### PR TITLE
OCM-902: Verify imdsv2 value is set post cluster creation

### PR DESCRIPTION
**Pull request by sunzhaohua2**
@xueli181114 PTAL
```
$ ginkgo -focus 63950 tests/e2e                                                              
time="2024-03-15T10:56:25+08:00" level=info msg="Running against env staging with gateway url https://api.stage.openshift.com"
time="2024-03-15T10:56:25+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-sts-pl : map[admin_enabled:false autoscale:false byok:false byovpc:true ccs:true channel_group:candidate cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge etcd_encryption:false fips:false imdsv2:optional kms_key_arn:false labeling:false major_version:4.14 multi_az:false oidc_config:managed private:true private_link:true product_id:rosa proxy:false region:us-east-1 sts:true tagging:false unified_acc_role_path: version: version_pattern:latest zones:]"
Running Suite: e2e tests suite - /Users/sunzhaohua/code/terraform-provider-rhcs/tests/e2e
=========================================================================================
Random Seed: 1710471380

Will run 1 of 60 specs
time="2024-03-15T10:56:25+08:00" level=warning msg="Got a global ENV variable RHCS_SOURCE set to terraform-redhat/rhcs. Going to replace all of the manifests files"
time="2024-03-15T10:56:25+08:00" level=warning msg="Got a global ENV variable RHCS_VERSION set to 1.5.1-prerelease.4. Going to replace all of the manifests files"
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
P [PENDING]
TF Test Verfication/Post day 1 tests Author:xueli-Critical-OCP-69145 @OCP-69145 @xueli Apply to change security group will be forbidden [NonHCPCluster]
/Users/sunzhaohua/code/terraform-provider-rhcs/tests/e2e/verfication_post_day1_test.go:261
------------------------------
SSStime="2024-03-15T10:56:26+08:00" level=info msg="Running terraform init against the dir /Users/sunzhaohua/code/terraform-provider-rhcs/tests/tf-manifests/rhcs/resource-import"
time="2024-03-15T10:56:27+08:00" level=info msg="Loaded cluster profile configuration from profile rosa-sts-pl : map[admin_enabled:false autoscale:false byok:false byovpc:true ccs:true channel_group:candidate cloud_provider:aws cluster_type:rosa-classic compute_machine_type:m5.xlarge etcd_encryption:false fips:false imdsv2:optional kms_key_arn:false labeling:false major_version:4.14 multi_az:false oidc_config:managed private:true private_link:true product_id:rosa proxy:false region:us-east-1 sts:true tagging:false unified_acc_role_path: version: version_pattern:latest zones:]"
•SSSSSSSSSSSSSSSSSSS

Ran 1 of 60 Specs in 3.883 seconds
SUCCESS! -- 1 Passed | 0 Failed | 1 Pending | 58 Skipped
PASS

Ginkgo ran 1 suite in 9.381973489s
Test Suite Passed
```

### CODE REVIEW

File: tests/e2e/verfication_post_day1_test.go [hunk 1]
```diff
@@ -336,5 +336,12 @@ var _ = Describe("TF Test", func() {
 				}
 			})
 		})
+		Context("Author:zhsun-Critical-OCP-63950 @zhsun", func() {
+			It("Verify imdsv2 value is set post cluster creation", ci.Day1Post, ci.Critical, func() {
+				if profile.Ec2MetadataHttpTokens != "" {
```
Comment (xueli181114) [inline tests/e2e/verfication_post_day1_test.go:341]: @sunzhaohua2 If it is empty, then it should have the default value of "optional" right?

Comment (sunzhaohua2) [reply tests/e2e/verfication_post_day1_test.go:341]: yes, will add a check

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**